omod/src/main/java/org/openmrs/module/providermanagement/fragment/controller/PersonNameFragmentController.java

```diff
@@ -14,13 +14,46 @@
 
 package org.openmrs.module.providermanagement.fragment.controller;
 
-import org.openmrs.layout.web.name.NameSupport;
+import java.lang.reflect.Method;
+import java.lang.reflect.InvocationTargetException;
+
 import org.openmrs.ui.framework.fragment.FragmentModel;
 
 public class PersonNameFragmentController {
-
-    public void controller(FragmentModel model) {
-        model.addAttribute("layoutTemplate", NameSupport.getInstance().getDefaultLayoutTemplate());
-    }
-
+	
+	/**
+	 * @param model
+	 * @throws ClassNotFoundException
+	 * @throws SecurityException
+	 * @throws NoSuchMethodException
+	 * @throws InvocationTargetException
+	 * @throws IllegalArgumentException
+	 * @throws IllegalAccessException
+	 */
+	
+	public void controller(FragmentModel model) throws ClassNotFoundException, NoSuchMethodException, SecurityException,
+	        IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+		
+		/*
+		 * backward compatibility
+		 */
+		Class<?> nameSurpport;
```

Comment:
```suggestion
		Class<?> nameSupport;
```

Reply:
hello @ibacher thank you for pointing them out let me try to solve them